src/App.tsx

```diff
@@ -1,24 +1,195 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorMessage } from './components/ErrorMesage/ErrorMesage';
+import { TodoForm } from './components/TodoForm/TodoForm';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { FilterType } from './types/FilterType';
+import { USER_ID } from './utils/fetchClient';
+import { TodoInfo } from './components/TodoInfo/TodoInfo';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [todoTitle, setTodoTitle] = useState('');
+  const [
+    currentFilter,
+    setCurrentFilter,
+  ] = useState<FilterType>(FilterType.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoadingTodo, setIsLoadingTodo] = useState(false);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        const todosFromServer = await getTodos(USER_ID);
+
+        setTodos(todosFromServer);
+      } catch {
+        setHasError(true);
+        setErrorMessage('Unable to update a todo');
+      }
+    };
+
+    fetchTodos();
+  }, []);
+
+  const addNewTodo = (title: string) => {
+    if (!title) {
+      setErrorMessage("Title can't be empty");
```

Comment:
Do you want to keep the code execution when there is no title? I think you need to add return statement here.